lib/puppet/type/service.rb

```diff
@@ -253,6 +255,49 @@ def sync
       end
     end
 
+    newproperty(:logonaccount, :required_features => :configurable_logon_credentials) do
+      desc "Specify an account for service logon on Windows"
+
+      munge do |value|
+        return value if value == 'LocalSystem'
+
+        # @account="Luchi",
+        # @account_type=:SidTypeUser,
+        # @domain="DESKTOP-P1OLRES",
+        # @domain_account="DESKTOP-P1OLRES\\Luchi",
+        # @sid="S-1-5-21-1825778692-957321060-2432639842-1000",
+        # @sid_bytes=[1, 5, 0, 0, 0, 0, 0, 5, 21, 0, 0, 0, 4, 44, 211, 108, 100, 143, 15, 57, 98, 35, 255, 144, 232, 3, 0, 0]>
+        account = Puppet::Util::Windows::SID.name_to_principal(value).domain_account
+        raise Puppet::Error.new(_"\"#{value}\" is not a valid account") if account.empty?
+        account
+      end
+
+      validate do |value|
+        raise Puppet::Error.new(_"This setting is only supported on Microsoft Windows.") unless Puppet::Util::Platform.windows?
+      end
+    end
+
+    newproperty(:logonpassword, :required_features => :configurable_logon_credentials) do
+      desc "Specify a password for service logon on Windows. Default value is an empty string when only the account is given."
+
+      validate do |value|
+        raise Puppet::Error.new(_"This setting is only supported on Microsoft Windows.") unless Puppet::Util::Platform.windows?
+        raise Puppet::Error.new(_"The 'logonaccount' parameter is mandatory.") unless @resource[:logonaccount]
+      end
+    end
+
+    newparam(:refreshservice, :required_features => :configurable_logon_credentials) do
+      desc "The service needs to be disabled and re-enabled in order to start using different logon credentials. By setting this
+        parameter to true, it will be done automatically. Default value is false."
+      
+      validate do |value|
+        raise Puppet::Error.new(_"This setting is only supported on Microsoft Windows.") unless Puppet::Util::Platform.windows?
+        raise Puppet::Error.new(_"The 'logonaccount' parameter is mandatory.") unless @resource[:logonaccount]
```

Comment:
There's no guarantee that the `resource[:logonaccount]` property will have been set prior to this parameter being validated. To validate across multiple properties, you'll want to move the [validate block to the type level](https://github.com/puppetlabs/puppet-specifications/blob/master/language/resource_types.md#validate) instead of param/property.